src/Bundle/EventListener/ReadListener.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\ResourceBundle\EventListener;
+
+use Sylius\Component\Resource\Context\Initiator\RequestContextInitiator;
+use Sylius\Component\Resource\Metadata\CreateOperationInterface;
+use Sylius\Component\Resource\Metadata\Operation\HttpOperationInitiator;
+use Sylius\Component\Resource\State\ProviderInterface;
+use Symfony\Component\HttpKernel\Event\RequestEvent;
+use Symfony\Component\HttpKernel\Exception\NotFoundHttpException;
+
+final class ReadListener
+{
+    public function __construct(
+        private HttpOperationInitiator $operationInitiator,
+        private RequestContextInitiator $contextInitiator,
```

Comment:
Here, I'll create an interface for the RequestContextInitiator 15 PRs after :dancers: 
I think I'll create an HttpOperationIniatorInterface too.